contrib/mpi-proxy-split/p2p_drain_send_recv.cpp

```diff
@@ -136,6 +136,14 @@ completePendingIrecvs()
         g_recvBytesByRank[worldRank] += call->count * size;
         bytesReceived += call->count * size;
         it = g_async_calls.erase(it);
+      } else {
+        // If the MPI_Test failed, we still update the iterator.
```

Comment:
You might want to update the comment for clarity. Here's a suggested comment:
 ```
 /* We update the iterator even if the MPI_Test fails.
  * Otherwise, the message we are waiting for will be sent
  * after the checkpoint. This can result in an infinite loop.
  *
  * NOTE: This function will be called only if the global arrays
  * do not match. This can happen if a second sender has sent
  * a message to us, and we will receive the message only
  * after the checkpoint.
```